File: testData/testMeGenerator/utilWithoutAccessableCtor/src/com/example/services/impl/Foo.java
package com.example.services.impl;

import com.example.warriers.FooFighter;
import com.example.foes.Fire;
<caret>
public class Foo {

    private static FooFighter fooFighter = new FearFighterImpl();

    public static String fight(Fire withFire,String foeName) {
        fooFighter.fight(withFire);
        return foeName;
    }
    
    private Foo() {

    }

}

File: testme-intellij-core/testData/testMeGenerator/scalaRequireExecutionContext/src/com/example/services/impl/Foo.scala
package com.example.services.impl
import com.example.common.LogSupport
import com.example.foes.Fire
import com.example.warriers.FooFighter

import scala.concurrent.{ExecutionContext, Future}


class Foo(fooFighter: FooFighter, implicit val ec:ExecutionContext) extends LogSupport{
  //  <caret>
  val withFire: Fire = new Fire()

  def fight(optFire:Option[String]) ={
    logger.info("it's started...")
    Future{
      fooFighter.fight(withFire)
    }
  }
}